### PR TITLE
#7188: Allow injection of FeatureEditor Toolbar buttons

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR allow injection of buttons in FeatureEditor grid from other plugins.
- It delegates  `DownloadLayer` button to the responsible plugin (the option in the toolbar has been removed, showIn, hideFrom or overrides can be used instead)
- Now multiple containers can be passed to the plugins containers. 

In draft because:
 - Waiting for branch stable relase
 - RFC about multiple containers entries. 
 - I'd like to remove Zoom all button, what do you think @tdipisa and reviewers? It is always disabled and it is not useful at all.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

Buttons for external plugins was hard-coded in feature grid

**What is the new behavior?**
- #7188
-  Now you can inject buttons into the feature grid.
- Bttons to create chart or to download are injected by the responsible plugins (  WidgetBuider and LayerDownload)
- They can be disabled by using `showIn`, `hideFrom` options
- You can inject multiple components for the same container. 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by tdipisa** [comment]
@offtherailz please change the milestone because in late September we will have the major release and we cannot add new features in master now. Therefore, if you need to merge something new in master, let's create the stable branch for the next major before.

@MV88 @dsuren1 @allyoucanmap @luorlandini @mbarto

File: web/client/utils/__tests__/PluginsUtils-test.js [hunk 1]
```diff
@@ -141,6 +141,53 @@ describe('PluginsUtils', () => {
...
+            };
+
+            const pluginsConfig = [{
+                name: "Test1"
+            }, "Container1", "Container2"];
+            const items = PluginsUtils.getPluginItems(defaultState, plugins, pluginsConfig, "Container1", "Container1", true, []);
+            expect(items.length).toBe(2);
+            expect(items.map(({id}) => id).includes(1, 2));
+        });
+        it('getPluginItems priority with arrays', () => {
```
Comment (mbarto) [inline web/client/utils/__tests__/PluginsUtils-test.js:164]: I would add a test where priorities are different

File: web/client/components/data/featuregrid/toolbars/Toolbar.jsx [hunk 2]
```diff
@@ -164,14 +150,7 @@ export default ({
...
             glyph="features-grid-set"/>
-        <TButton
-            id="grid-map-chart"
-            keyProp="grid-map-chart"
-            tooltipId="featuregrid.toolbar.createNewChart"
-            disabled={disableToolbar}
-            visible={mode === "VIEW" && showChartButton}
-            onClick={events.chart}
-            glyph="stats"/>
+        {toolbarItems.map(({Component}) => <Component {...props} disabled={disableToolbar} />)}
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/components/data/featuregrid/toolbars/Toolbar.jsx:153]: There is an bug that puts one button at the end when the tooltip is active (sync with map). 
Anyway I'm going improve and explicit a little bit more sorting, in order to allow injection in every position.

**Review by mbarto — APPROVED** [review]

**Comment by offtherailz** [comment]
note: backport have to be scheduled after the release as for assigned milestone @tdipisa

**Pull request merged**